### PR TITLE
Handle commas for BubbleTextField (fixes #2427)

### LINKED ISSUE #2427: Cannot type Cyrillic "Б" in the address field of mail editor
It is on the same key as US comma (`,`) and we probably check keycode and don't insert anything. We shouldn't use keycode in this case.

### PR DESCRIPTION
**Pull request by SnowyMouse**
This fixes #2427 by using onkeypress instead of onkeydown, which lets us intercept keys while knowing the actual value of the character being inserted.

I tested this by changing my system's keyboard layout to one that doesn't have the U.S. comma key code correspond to a comma.

### CODE REVIEW

File: src/gui/base/TextField.js [hunk 1]
```diff
@@ -51,6 +51,7 @@ export class TextField {
 	onblur: Stream<*>;
 	skipNextBlur: boolean;
 	_keyHandler: ?keyHandler; // interceptor used by the BubbleTextField to react on certain keys
+	_keyPressHandler: ?Function; // used by BubbleTextField to react on commas
```
Comment (charlag) [inline src/gui/base/TextField.js:54]: Don't use `Function`, it's actually `any`, let it be `(KeyboardEvent) => mixed`

**Review by charlag — COMMENTED** [review]
Thanks, we'll have a look a little bit later!

**Comment by SnowyMouse** [comment]
I've updated it with your suggestion.

**Review by charlag — COMMENTED** [review]
MDN shows big red warning that `keypress` is deprecated:
https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onkeypress

I think we can just switch to `KeyboardEvent.key` like you did without another event handler?
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key

**Comment by SnowyMouse** [comment]
I see. That makes sense.

I added "key" to the KeyPress type and used the keydown event as it was doing before.

**Review by charlag — APPROVED** [review]
Works for me with both Cyrillic and US layouts, thanks a lot, this is a great fix

**Pull request merged**